cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/DefaultStoreTest.java

```diff
@@ -1061,6 +1059,40 @@ public void testRunningInRangeMulti() throws Exception {
                         runIdsToTime(store.getRunningInRange(1, 45 * 10)));
   }
 
+  @Test
+  public void testDeleteWorkflowStatistics() throws Exception {
+    String namespaceName = "namespace1";
+    String appName = "app1";
+    String workflowName = "workflow1";
+    String mapReduceName = "mapReduce1";
+
+    ApplicationId appId = Ids.namespace(namespaceName).app(appName);
+    ProgramId mapReduceProgram = appId.mr(mapReduceName);
+
+    long currentTime = System.currentTimeMillis();
+    String workflowRunId = RunIds.generate(currentTime).getId();
+    ProgramRunId workflowRun = appId.workflow(workflowName).run(workflowRunId);
+
+    // start Workflow
+    store.setStart(workflowRun.getParent().toId(), workflowRun.getRun(), currentTime);
+
+    // start MapReduce as a part of Workflow
+    Map<String, String> systemArgs = ImmutableMap.of(ProgramOptionConstants.WORKFLOW_NODE_ID, mapReduceName,
+                                                     ProgramOptionConstants.WORKFLOW_NAME, workflowName,
+                                                     ProgramOptionConstants.WORKFLOW_RUN_ID, workflowRunId);
+
+    RunId mapReduceRunId = RunIds.generate(currentTime + 10);
+    store.setStart(mapReduceProgram.toId(), mapReduceRunId.getId(), currentTime + 10, null,
+                   ImmutableMap.<String, String>of(), systemArgs);
+
+    // stop the MapReduce program
+    store.setStop(mapReduceProgram.toId(), mapReduceRunId.getId(), currentTime + 50, ProgramRunStatus.COMPLETED);
+
+    // stop Workflow
+    store.setStop(workflowRun.getParent().toId(), workflowRun.getRun(), currentTime + 110, ProgramRunStatus.COMPLETED);
+
```

Comment:
I don't see how this is testing what you implemented.
The code you implemented only gets executed upon app delete, right?